### PR TITLE
Fix some vmi cleanup issues in vm_networking.go

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Ensure that some tests in vmi_networking.go clean up the VMIs they
create. The tests affected by this were partly assuming a complete
cleanup and partly assuming that they share common VMIs.
As a result they were anyway already creating their own VMIs, but were
not cleaning up VMIs from the previous test runs.

An example is this run: https://prow.apps.ovirt.org/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5248/pull-kubevirt-e2e-k8s-1.19/1372226042629984256.

The [overview.log](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5248/pull-kubevirt-e2e-k8s-1.19/1372226042629984256/artifacts/k8s-reporter/1_overview.log) shows the accumulated VMIs clearly:

```
kubevirt-test-default1   pod/virt-launcher-testvmi-5hrxg-wr4cz         0/2     Terminating   0          6m      10.244.140.80    node02   <none>           <none>
kubevirt-test-default1   pod/virt-launcher-testvmi-6prdr-wdtsl         2/2     Running       0          15m     10.244.140.116   node02   <none>           <none>
kubevirt-test-default1   pod/virt-launcher-testvmi-8qlr7-8r9g6         2/2     Running       0          19m     10.244.140.121   node02   <none>           <none>
kubevirt-test-default1   pod/virt-launcher-testvmi-fjbz9-hcwwf         2/2     Running       0          17m     10.244.140.126   node02   <none>           <none>
kubevirt-test-default1   pod/virt-launcher-testvmi-hm98m-kmd69         2/2     Running       0          13m     10.244.140.79    node02   <none>           <none>
kubevirt-test-default1   pod/virt-launcher-testvmi-m67st-fl2c2         2/2     Running       0          19m     10.244.140.76    node02   <none>           <none>
kubevirt-test-default1   pod/virt-launcher-testvmi-nqn77-fbpc7         2/2     Running       0          16m     10.244.140.119   node02   <none>           <none>
kubevirt-test-default1   pod/virt-launcher-testvmi-t4m6n-5hlm6         0/2     Terminating   0          6m      10.244.140.77    node02   <none>           <none>
```

There may be more cleanup issues, but this resolves the most basic left-over issues.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
@AlonaKaplan FYI

File: tests/network/vmi_networking.go [hunk 1]
```diff
@@ -818,9 +811,7 @@ var _ = SIGDescribe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com]
 			}
 
 			BeforeEach(func() {
-				if !tests.HasLiveMigration() {
```
Comment (vladikr) [inline tests/network/vmi_networking.go:821]: Perhaps, we just need to move all migration related tests to the migration_test.go

File: tests/network/vmi_networking.go [hunk 2]
```diff
@@ -818,9 +811,7 @@ var _ = SIGDescribe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com]
...
-					Skip("LiveMigration feature gate is not enabled in kubevirt-config")
-				}
+				tests.SkipIfMigrationIsNotPossible()
```
Comment (vladikr) [inline tests/network/vmi_networking.go:814]: :+1:

**Review by fgimenez — APPROVED** [review]
/lgtm

File: tests/network/vmi_networking.go (hunk 1 above)
Comment (rmohr) [reply tests/network/vmi_networking.go:821]: probably a `migration` package with various focused files related to storage, network various vm-configurations?

Comment (vladikr) [reply tests/network/vmi_networking.go:821]: we could just consolidate all of these tests in tests/migration_test.go

File: tests/utils.go [hunk 3]
```diff
@@ -3850,6 +3850,22 @@ func SkipIfOpenShift4(message string) {
 	}
 }
 
+func SkipIfMigrationIsNotPossible() {
```
Comment (AlonaKaplan) [inline tests/utils.go:3853]: Should be used [here](https://github.com/kubevirt/kubevirt/blob/d6f4ff68461474c7889a8c601e8efe18a4e2f8fd/tests/vmi_multus_test.go) as well.

**Review by AlonaKaplan — COMMENTED** [review]
I'm ok with the change but I don't understand how it is related to the timeout in the description of the PR.
Can you please explain/give an example of a flow causing the timeout?

**Comment by rmohr** [comment]
> I'm ok with the change but I don't understand how it is related to the timeout in the description of the PR.
> Can you please explain/give an example of a flow causing the timeout?

Your suspicion that it may  be unrelated is probably correct. It may be more related to some pod startup delays, where some hints are in the journal log.

I will rephrase the PR description, so that this can go in, if that is fine.

File: tests/utils.go (hunk 3 above)
Comment (rmohr) [reply tests/utils.go:3853]: Addressed.

**Pull request merged**